src/slurm_plugin/slurm_resources.py

```diff
@@ -414,7 +427,31 @@ def is_backing_instance_valid(self, log_warn_if_unhealthy=True):
                         self,
                         self.state_string,
                     )
-                return False
+                # Allow a few iterations for the eventual consistency of EC2 data
+                logger.info(
+                    f"ec2_backing_instance_max_count {ec2_backing_instance_max_count} "
+                    f"nodes_without_backing_instance_count_map {nodes_without_backing_instance_count_map}"
+                )
+                if any(
+                    args in [None]
+                    for args in [ec2_backing_instance_max_count, nodes_without_backing_instance_count_map]
+                ):
+                    logger.info("No max count or map provided, ignoring backing instance timeout")
+                    return False
+                elif nodes_without_backing_instance_count_map.get(self.name, 0) >= ec2_backing_instance_max_count:
+                    logger.warning(f"Instance {self.name} availability has timed out.")
+                    return False
+                else:
+                    if not self.missing_count_incremented:
+                        nodes_without_backing_instance_count_map[self.name] = (
+                            nodes_without_backing_instance_count_map.get(self.name, 0) + 1
+                        )
+                        logger.warning(
+                            f"Instance {self.name} is not yet available in EC2, "
```

Comment:
This is similar to "Node state check: no corresponding instance in EC2 for node %s, node state: %s" at line 426.
Is it possible to keep just one? I would be inclined to keep what's already present, at most by adding other details, rather than changing it completely.

Reply:
I agree that is does end up duplicating the info, but we don't know if the missing count will be incremented before the if block.  I could remove the first part